src/L2/discounts/TalentProtocolDiscountValidator.sol

```diff
@@ -0,0 +1,62 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import {Ownable} from "solady/auth/Ownable.sol";
+
+import {IDiscountValidator} from "src/L2/interface/IDiscountValidator.sol";
+
+/// @title Discount Validator for: Talent Protocol Builder Score
+///
+/// @notice Enables discounts for holders of Talent Protocol Builder Score tokens. Discounts are granted
+///         Based on the claimer having some score higher than this contract's `threshold`.
+///
+/// @author Coinbase (https://github.com/base-org/usernames)
+contract TalentProtocolDiscountValidator is IDiscountValidator, Ownable {
+    /// @notice Thrown when setting a critical address to the zero-address.
+    error NoZeroAddress();
+
+    /// @notice Interface with the Talent Protocol `PassportBuilderScore` contract.
+    TalentProtocol immutable talentProtocol;
+
+    /// @notice The score threshold for qualifying for a discount.
+    uint256 public threshold;
+
+    /// @notice constructor
+    constructor(address owner_, address talentProtocol_, uint256 threshold_) {
+        if (owner_ == address(0)) revert NoZeroAddress();
+        if (talentProtocol_ == address(0)) revert NoZeroAddress();
+        talentProtocol = TalentProtocol(talentProtocol_);
+        threshold = threshold_;
+        _initializeOwner(owner_);
+    }
+
+    /// @notice Allows the `owner` to set the qualifying threshold.
+    ///
+    /// @param threshold_ The new discount-qualifying threshold.
+    function setThreshold(uint256 threshold_) external onlyOwner {
+        threshold = threshold_;
+    }
+
+    /// @notice Required implementation for compatibility with IDiscountValidator.
+    ///
+    /// @dev No additional data is required for validating this discount.
+    ///
+    /// @param claimer the discount claimer's address.
+    ///
```

Comment:
Should the natspec still note the unused bytes calldata param other than in the `/// @dev` comment?

Reply:
IDK, i don't think so. It appropriately references the inherited natspec in IDiscountValidator and then describes why it doesn't need `bytes validationData`. If you want to suggest an edit, I'll consider it though :)

Reply:
If you don't think so that's good with me! Just wanted to ensure it wasn't an oversight.